pkgs/development/compilers/elm/default.nix

```diff
@@ -91,9 +91,29 @@ let
           inherit nodejs pkgs;
           inherit (stdenv.hostPlatform) system;
         };
-    in with hsPkgs.elmPkgs; {
+    in with hsPkgs.elmPkgs; rec {
       elm-test = patchBinwrap [elmi-to-json] nodePkgs.elm-test;
       elm-verify-examples = patchBinwrap [elmi-to-json] nodePkgs.elm-verify-examples;
+      elm-coverage =
+        let patched = patchBinwrap [elm elmi-to-json] nodePkgs.elm-coverage;
+        in patched.override {
+          preRebuild = ''
+            sed 's/\"install\".*/\"install\":\"echo no-op\"/g' --in-place package.json
+
+            # This should not be needed (thanks to binwrap* being nooped) but for some reason it still needs to be done
+            # in case of just this package
+            sed 's/\"install\".*/\"install\":\"echo no-op\",/g' --in-place node_modules/elmi-to-json/package.json
```

Comment:
This is the biggest hack and I can't really explain why is needed.

Reply:
Probably since it's needed for elm-test, which is installer here as a dependency?

Reply:
That's the case of `elm-verify-examples` as well and in that case the `writeScriptBin` works as expected. I have a bad feeling that this somewhat depends on how `PATH` is being defined and potentially in case of this package the version from `node_modules/.bin` takes precedence over the script. That would be very unfortunate behavior. If that's the case we need a better way to patch this.

`binwrap` package also provides library functions which can (and are) called from packages code (and hopefully should just check the binaries are already present and do nothing). I think if we would try to avoid installing that package in any way we would need to do a lot of patching in source files. One thing that comes to mind is to let npm install the library but remove its bin files.

Some part of me always wants to start crying every time I start considering doing hacks like this but so far I don't have idea about how to solve these issues in more robust way.

Reply:
Yeah most probably need to check $PATH precedence for node_modules, but I think this is good enough to work.